packages/qwik-city/runtime/src/qwik-city-component.tsx

```diff
@@ -263,7 +263,7 @@ export const QwikCityProvider = component$<QwikCityProps>((props) => {
         }
         const newHref = pageData.href;
         const newURL = new URL(newHref, trackUrl);
-        if (!isSamePathname(newURL, trackUrl)) {
+        if (!isSamePath(newURL, trackUrl)) {
           trackUrl = newURL;
```

Comment:
because of this line,

```js
if (!isSamePathname(newURL, trackUrl)) {
          trackUrl = newURL;
          loadRoutePromise = loadRoute(qwikCity.routes, qwikCity.menus, qwikCity.cacheModules, trackUrl.pathname);
        }

        if(trackUrl.href !== newURL.href) {
          trackUrl = newURL;
        }
```

it would be better if it only reassigns this value `trackUrl`.
because it does need to recall loadRoute function
how do you think so

Reply:
@JerryWu1234 not sure what you mean here. This PR just seems to change the check for equality so that it also checks the search string to decide when to re-run route loaders. I don't see the `trackUrl.href` check?

Reply:
https://github.com/QwikDev/qwik/blob/a219b51f701bf29e8782dbbf5c0ada0928f34f39/packages/qwik-city/runtime/src/client-navigate.ts#L19

it is the reason why route didn't redirect.
So if newURL and  trackUrl aren't equal, we can only make `trackUrl ` get new URL.

so that [samePath](https://github.com/QwikDev/qwik/blob/a219b51f701bf29e8782dbbf5c0ada0928f34f39/packages/qwik-city/runtime/src/client-navigate.ts#L14C5-L14C49) would become true ,and route will redirect.
if so that, it doesn't need to call `loadRoute ` 

@wmertens

Reply:
I think the correct behavior is to re run the loader, we are doing the same with `actions` so for consistency I will prefer call `loadRoute`